ApollosProject/apollos-apps
# Chore: Analytics refactor

## ✏️ Description



Significantly refactors/simplifies the ui-analytics package and implements built-in support for a global Segment setup.

### Backstory
In evaluating our data needs, I've learned that:

- We need to track much better data for our own purposes. We have no cross-church analytics and no way to look at app performance in the macro
- This not only makes product decisions harder, but leaves all of our teams (marketing, sales, customer success, etc) without access to valuable shared data
- Furthermore, many of our clients are starting to desire more data sent to more platforms: Tableau, GA, Rock, etc. Setting up additional data destinations is difficult with our current setup
- Something like Astronomer is not optimized for click-stream data (although it works great in a reverse ETL setup like we have it now)

Also, `identify` is never being called on production, so even our current analytics tools is not getting that information.

### Customer Data Platform
I am still evaluating Customer Data Platform solutions. Segment, mParticle, Rudderstack, Salesforce, etc. Segment appears to be a clear choice for now:

- Price model nets it out cheaper then some competitors with lots of room to grow
- High number of integrations
- Ability to programmatic control destinations and filtering per-destination, meaning we could use Segment to funnel customer-specific data to customer-specific tools (such as use Segment to funnel Rock data for a single church into their Rock)

### ~~This removes our server-side analytics tracking~~
~~So far, this PR removes tracking to our server-side analytics. I'm not even sure if any church is using this outside of Crossroads (which, unfortunately, they use Segment....and the greatest drawback of Segment is they don't seem to have a Segment to Segment solution). However, these track events are accounting for a reasonable amount of our web traffic on the API, so removing these hits to our API might improve performance overall.~~

### ~~I made the TV Changes blind~~
~~I tested mobile, but I have not booted TV to see if these changes work without error. I believe they should, but will need more 👀 on this.~~

## 🔬 To Test
1. Add Segment key to your .env: `SEGMENT_KEY=rwkfWr0HpIW4eSUZqI30BmcMAdwsVrW0` (I don't anticipate using this key in production)
2. Boot up the sim, click around
3. Login to Segment and check [the live event stream](https://app.segment.com/differential-conrad/sources/apollos/debugger) (password in 1password)

## 📸 Screenshots

<img width="611" alt="Screenshot 2022-11-22 at 11 50 12 PM" src="https://user-images.githubusercontent.com/103927/203480254-658c4da3-b7db-498c-8fe7-cb779c3688a2.png">




## ⚠️ To-do Before Merge

- [ ] Test that amplitude and onesignal events are still coming through correctly
- [ ] Test that interactions (like completing a content item) is going into the API correctly



## 📝 Checklist
